packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/RealmSyncUtils.kt

```diff
@@ -122,14 +118,16 @@ internal fun convertSyncErrorCode(syncError: SyncErrorCode): SyncException {
             // See https://github.com/realm/realm-core/blob/master/src/realm/sync/protocol.hpp#L217
             // Use https://docs.google.com/spreadsheets/d/1SmiRxhFpD1XojqCKC-xAjjV-LKa9azeeWHg-zgr07lE/edit
             // as guide for how to categorize Session type errors.
-            when (syncError.code) {
+            when (errorCode.code) {
                 ProtocolSessionErrorCode.RLM_SYNC_ERR_SESSION_BAD_QUERY -> { // Flexible Sync Query was rejected by the server
                     BadFlexibleSyncQueryException(message)
                 }
                 ProtocolSessionErrorCode.RLM_SYNC_ERR_SESSION_PERMISSION_DENIED ->
                     // Permission denied errors should be unrecoverable according to Core, i.e. the
                     // client will disconnect sync and transition to the "inactive" state
                     UnrecoverableSyncException(message)
+                ProtocolSessionErrorCode.RLM_SYNC_ERR_SESSION_COMPENSATING_WRITE ->
+                    CompensatingWriteException(message, syncError!!.compensatingWrites)
```

Comment:
I guess `syncError` will not always be non-null!?
```suggestion
                    CompensatingWriteException(message, syncError.compensatingWrites)
```